# Add const qualifier to pam_handle_t* argument of pam_strerror(),
# pam_getenv(), pam_getenvlist(), pam_prompt() and pam_vprompt().

--- Linux-PAM/libpam/include/security/_pam_types.h
+++ Linux-PAM/libpam/include/security/_pam_types.h
@@ -175,16 +175,16 @@ extern int PAM_NONNULL((1))
 pam_get_item(const pam_handle_t *pamh, int item_type, const void **item);
 
 extern const char *
-pam_strerror(pam_handle_t *pamh, int errnum);
+pam_strerror(const pam_handle_t *pamh, int errnum);
 
 extern int PAM_NONNULL((1,2))
 pam_putenv(pam_handle_t *pamh, const char *name_value);
 
 extern const char * PAM_NONNULL((1,2))
-pam_getenv(pam_handle_t *pamh, const char *name);
+pam_getenv(const pam_handle_t *pamh, const char *name);
 
 extern char ** PAM_NONNULL((1))
-pam_getenvlist(pam_handle_t *pamh);
+pam_getenvlist(const pam_handle_t *pamh);
 
 /* ---------- Common Linux-PAM application/module PI ----------- */
 
--- Linux-PAM/libpam/include/security/pam_ext.h
+++ Linux-PAM/libpam/include/security/pam_ext.h
@@ -59,11 +59,11 @@ extern void PAM_FORMAT((printf, 3, 4)) PAM_NONNULL((3))
 pam_syslog (const pam_handle_t *pamh, int priority, const char *fmt, ...);
 
 extern int PAM_FORMAT((printf, 4, 0)) PAM_NONNULL((1,4))
-pam_vprompt (pam_handle_t *pamh, int style, char **response,
+pam_vprompt (const pam_handle_t *pamh, int style, char **response,
 	     const char *fmt, va_list args);
 
 extern int PAM_FORMAT((printf, 4, 5)) PAM_NONNULL((1,4))
-pam_prompt (pam_handle_t *pamh, int style, char **response,
+pam_prompt (const pam_handle_t *pamh, int style, char **response,
 	    const char *fmt, ...);
 
 #define pam_error(pamh, fmt...) \
--- Linux-PAM/libpam/pam_env.c
+++ Linux-PAM/libpam/pam_env.c
@@ -22,7 +22,7 @@
 /* helper functions */
 
 #ifdef DEBUG
-static void _pam_dump_env(pam_handle_t *pamh)
+static void _pam_dump_env(const pam_handle_t *pamh)
 {
     int i;
 
@@ -282,7 +282,7 @@ int pam_putenv(pam_handle_t *pamh, const char *name_value)
  *  Return the value of the requested environment variable
  */
 
-const char *pam_getenv(pam_handle_t *pamh, const char *name)
+const char *pam_getenv(const pam_handle_t *pamh, const char *name)
 {
     int item;
 
@@ -316,7 +316,7 @@ const char *pam_getenv(pam_handle_t *pamh, const char *name)
     }
 }
 
-static char **_copy_env(pam_handle_t *pamh)
+static char **_copy_env(const pam_handle_t *pamh)
 {
     char **dump;
     int i = pamh->env->requested;          /* reckon size of environment */
@@ -355,7 +355,7 @@ static char **_copy_env(pam_handle_t *pamh)
     return dump;
 }
 
-char **pam_getenvlist(pam_handle_t *pamh)
+char **pam_getenvlist(const pam_handle_t *pamh)
 {
     int i;
 
--- Linux-PAM/libpam/pam_strerror.c
+++ Linux-PAM/libpam/pam_strerror.c
@@ -33,7 +33,7 @@
 
 #include "pam_private.h"
 
-const char *pam_strerror(pam_handle_t *pamh UNUSED, int errnum)
+const char *pam_strerror(const pam_handle_t *pamh UNUSED, int errnum)
 {
     switch (errnum) {
     case PAM_SUCCESS:
--- Linux-PAM/libpam/pam_vprompt.c
+++ Linux-PAM/libpam/pam_vprompt.c
@@ -46,7 +46,7 @@
 #include "pam_private.h"
 
 int
-pam_vprompt (pam_handle_t *pamh, int style, char **response,
+pam_vprompt (const pam_handle_t *pamh, int style, char **response,
 	     const char *fmt, va_list args)
 {
   struct pam_message msg;
@@ -101,7 +101,7 @@ pam_vprompt (pam_handle_t *pamh, int style, char **response,
 }
 
 int
-pam_prompt (pam_handle_t *pamh, int style, char **response,
+pam_prompt (const pam_handle_t *pamh, int style, char **response,
 	    const char *fmt, ...)
 {
   va_list args;
